common/common.go

```diff
@@ -461,3 +461,23 @@ func SetOptionalRedisPasswordFromKubeConfig(ctx context.Context, kubeClient kube
 	redisOptions.Password = string(secret.Data[RedisInitialCredentialsKey])
 	return nil
 }
+
+const (
+	DefaultRedisCompressionKey = "redis.compression"
+)
+
+func GetRedisCompressionFromKubeConfig(ctx context.Context, kubeClient kubernetes.Interface, namespace string) (string, error) {
```

Comment:
would be nice to document this change, also add details how to get correct compression type from your redis server, maybe default values for different redis and argocd versions